pinot-common/src/test/java/org/apache/pinot/common/metrics/AbstractMetricsTest.java

```diff
@@ -48,4 +53,30 @@ public void testAddOrUpdateGauge() {
     controllerMetrics.removeGauge(metricName);
     Assert.assertTrue(controllerMetrics.getMetricsRegistry().allMetrics().isEmpty());
   }
+
+  @Test
+  public void testUpdateAndRemoveGauge()
```

Comment:
In order to avoid leaving the orphan metrics from the previous test, can we add a method with the `@BeforeTest` annotation to clean up all the old metrics?

You can do sth like:
```
    // Remove metrics if exists
    for (Map.Entry<PinotMetricName, PinotMetric> entry : metricsRegistry.allMetrics().entrySet()) {
      metricsRegistry.removeMetric(entry.getKey());
    }
```

Reply:
We are creating a new `YammerMetricsRegistry` every test run. So we should be good I suppose.

Reply:
I think we could even remove this test. The amount of code added in the test is likely not worth it. But LGTM. Thanks @tibrewalpratik17 for going out of your way to fix day to day bugs.